core/src/main/java/io/questdb/griffin/engine/functions/bool/InSymbolFunctionFactory.java

```diff
@@ -113,20 +134,58 @@ public boolean getBool(Record rec) {
         @Override
         public void init(SymbolTableSource symbolTableSource, SqlExecutionContext executionContext) throws SqlException {
             arg.init(symbolTableSource, executionContext);
+            if (deferredValues != null) {
+                for (int i = 0, n = deferredValues.size(); i < n; i++) {
+                    deferredValues.getQuick(i).init(symbolTableSource, executionContext);
+                }
+            }
+
             final StaticSymbolTable symbolTable = arg.getStaticSymbolTable();
             if (symbolTable != null) {
                 intSet.clear();
                 for (int i = 0, n = set.size(); i < n; i++) {
                     intSet.add(symbolTable.keyOf(set.get(i)));
                 }
+                if (deferredValues != null) {
+                    for (int i = 0, n = deferredValues.size(); i < n; i++) {
+                        final Function func = deferredValues.getQuick(i);
+                        final int typeTag = ColumnType.tagOf(func.getType());
+                        if (typeTag == ColumnType.CHAR) {
+                            sink.clear();
+                            sink.put(func.getChar(null));
+                            intSet.add(symbolTable.keyOf(sink));
+                        } else {
+                            intSet.add(symbolTable.keyOf(func.getStr(null)));
+                        }
+                    }
+                }
                 testFunc = intTest;
             } else {
+                if (deferredValues != null) {
+                    deferredSet.clear();
+                    for (int i = 0, n = deferredValues.size(); i < n; i++) {
+                        final Function func = deferredValues.getQuick(i);
+                        final int typeTag = ColumnType.tagOf(func.getType());
+                        if (typeTag == ColumnType.CHAR) {
+                            deferredSet.add(String.valueOf(func.getChar(null)));
```

Comment:
Note: here we allocate on the query execution path. I'll push a patch to fix that tomorrow.

Reply:
Simplified the function in a36593b43a5973c3e16d0b6279e124e0bc089a1f

Now it supports bind variables of only string type which is fine considering that it's the right way to define a symbol bind variable.